pkg/pillar/hypervisor/hypervisor.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2017-2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package hypervisor
+
+import (
+	"os"
+)
+
+// Hypervisor provides methods for manipulating domains on the host
+type Hypervisor interface {
+	Name() string
+
+	Create(string, string) (int, error)
+
+	Start(string, int) error
+	Tune(string, int, int) error
+	Stop(string, int, bool) error
+	Delete(string, int) error
+	Info(string, int) error
+	LookupByName(string, int) (int, error)
+
+	IsDeviceModelAlive(int) bool
+
+	PCIReserve(string) error
+	PCIRelease(string) error
+}
+
+// GetHypervisor returns a particular hypervisor implementation
+func GetHypervisor(hint string) Hypervisor {
+	var knownHypervisors = map[string]func() Hypervisor{
```

Comment:
This should be move out into a package level `var` rather than in a `func` invocation

Reply:
Hm. I'm not sure I agree -- I actually see it as information hiding -- there's no reason for the rest of the methods to know about this kind of mapping.

Are you worried about anything in particular here @deitch ?

Reply:
Will this never be used anywhere else? Recreating the map each time we make a call to `GetHypervisor` could lead someone in the future to start to modify or change it on the fly. If it is package level `var`, it indicates by itself that "this is something unchangeable"

Reply:
Now that I had to implement a 2nd method that uses it -- I actually moved it out.